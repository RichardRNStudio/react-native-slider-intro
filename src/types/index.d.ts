export * from './AnimatedValues.types';
export * from './Button.types';
export * from './SliderIntroItem.types';
export * from './Slide.types';
export * from './SliderIntro.types';
export * from './StatusBar.types';
