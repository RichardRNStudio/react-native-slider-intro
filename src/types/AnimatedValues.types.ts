import { Animated } from 'react-native';

export type AnimatedValues = {
  _moveSlideTranslateX: Animated.Value;
  _slideDotScaleX: Animated.Value;
  _slideDotTranslateX: Animated.Value;
  _opacityOfNextButton: Animated.Value;
  _opacityOfDoneButton: Animated.Value;
  _opacityOfSkipButton: Animated.Value;
};
