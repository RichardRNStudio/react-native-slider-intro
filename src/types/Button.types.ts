export enum ButtonType {
  Skip = 'skip',
  Done = 'done',
  Next = 'next',
  Prev = 'prev',
}
export interface ButtonProps {
  type: ButtonType;
  label?: string;
}
