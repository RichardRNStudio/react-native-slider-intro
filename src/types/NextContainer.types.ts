import type { Slide } from './Slide.types';

export interface NextContainerProps {
  goToNewSlide: (slideIndex: number) => void;
  isLastSlide: boolean;
  animations: Slide['animations'];
}
