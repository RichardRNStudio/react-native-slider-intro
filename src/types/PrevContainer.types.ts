export interface PrevContainerProps {
  goToNewSlide: (slideIndex: number) => void;
  setDefaultState: () => void;
}
