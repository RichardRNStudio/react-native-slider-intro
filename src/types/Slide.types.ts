export type Slide = {
  active: number;
  previous: number;
  marginLeft: number;
  dotMarginLeft: number;
  expectOpacityOfNext: number;
  expectOpacityOfDone: number;
  expectOpacityOfSkip: number;
};
