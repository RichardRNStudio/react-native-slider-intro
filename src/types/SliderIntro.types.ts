import React from 'react';
import type { ColorValue } from 'react-native';
import type { SliderIntroItemProps } from './SliderIntroItem.types';
import type { ButtonType } from './Button.types';

export enum BackHandlerType {
  ActiveMinusOne = 'activeMinusOne',
  Previous = 'previous',
}

type LeftButtonType = 'previous' | ButtonType.Skip;

export type SliderIntroProps = {
  /**
   * Default render - array of items. Children will be ignored if data is passed.
   */
  data: SliderIntroItemProps[];
  /**
   * Custom render - if children is passed, data will be ignored. numberOfSlides is required in this case.
   */
  numberOfSlides: number;
  /**
   * Custom render - children elements to render.
   */
  children: React.ReactNode;
  navigationBarBottom: number;
  navigationBarHeight: number;
  animateSlideSpeed: number;
  navContainerMaxSizePercent: number;
  dotWidth: number;
  fixDotOpacity: number;
  fixDotBackgroundColor: ColorValue;
  animatedDotBackgroundColor: ColorValue;
  animateDotSpeed: number;
  animateDotBouncing: number;
  skipLabel: string;
  nextLabel: string;
  doneLabel: string;
  showLeftButton: boolean;
  leftButtonType: LeftButtonType;
  columnButtonStyle: boolean;
  limitToSlide: number;
  renderSkipButton?: (skipLabel: string) => React.ReactNode;
  renderNextButton?: (nextLabel: string) => React.ReactNode;
  renderDoneButton?: (doneLabel: string) => React.ReactNode;
  onSkip?: () => void;
  onDone?: () => void;
};
