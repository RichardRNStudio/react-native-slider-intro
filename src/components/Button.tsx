import React from 'react';
import { StyleSheet, Text, View } from 'react-native';
import { ButtonType, type ButtonProps } from '../types/Button.types';

const styles = StyleSheet.create({
  skipButton: {
    justifyContent: 'center',
    alignItems: 'center',
  },
  label: {
    textTransform: 'uppercase',
    fontWeight: 'bold',
    color: 'white',
    fontSize: 14,
  },
});

const Button = ({ label, type }: ButtonProps) => {
  if (type === ButtonType.Skip) {
    return (
      <View style={styles.skipButton}>
        <Text style={styles.label}>{label}</Text>
      </View>
    );
  }
  return <Text style={styles.label}>{label}</Text>;
};

export default Button;
