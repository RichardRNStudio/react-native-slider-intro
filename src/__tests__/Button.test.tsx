import React from 'react';
import renderer from 'react-test-renderer';
import Button from '../components/Button';
import { Text } from 'react-native';
import { ButtonType } from '../types/Button.types';

describe('Button', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<Button label="test" type={ButtonType.Done} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('displays the correct label', () => {
    const component = renderer.create(
      <Button label="test" type={ButtonType.Done} />
    );
    const instance = component.root;
    const textComponent = instance.findByType(Text);
    expect(textComponent.props.children).toBe('test');
  });
});
