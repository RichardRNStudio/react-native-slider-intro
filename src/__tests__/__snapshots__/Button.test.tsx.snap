// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders correctly 1`] = `
<Text
  style={
    {
      "color": "white",
      "fontSize": 14,
      "fontWeight": "bold",
      "textTransform": "uppercase",
    }
  }
>
  test
</Text>
`;
